Hotfix/hotfix 030119 general cleanup

jomurgel opened the pull request:
## DESCRIPTION
This fixes a couple of minor issues with semantics and comments.

First replaces any instance of the `<!-- .grid-x -->` with the appropriate terminating comment.
- https://github.com/WebDevStudios/wd_s/commit/27a8c3eda60ee90906765fa1fe4b50165bd038eb

Next, I've added a new function [_s_display_hero_heading()](https://github.com/WebDevStudios/wd_s/blob/qa/hotfix-030119-general-cleanup/inc/acf.php#L416) which outputs the title for the hero. If it is the first item on the page and isn't the front page or home index, it will output an h1, else output an h2. This is important since the hero is quite often used as the top/header element on ACF pages.
- https://github.com/WebDevStudios/wd_s/commit/7623b6241edbcdd703027cbcd81feb6d0061e1de

Finally, there were a few inconsistencies and issues with ACF content blocks padding having or not having padding depending on the options. This small update has been made to keep padding ONLY on elements without a background.
- https://github.com/WebDevStudios/wd_s/commit/dca1601693b028b51974901804f954eb4541adf4

jomurgel committed:
runs audit and fixes npm

jomurgel committed:
updates terminating comment

jomurgel committed:
adds function to return h1/h2 in hero based on location.

jomurgel committed:
resolves global content block issues with padding/margin

jomurgel committed:
builds

jomurgel committed:
adds doc block

coreymcollins commented on assets/sass/modules/content-blocks/_global.scss:
```diff
@@ -14,6 +14,7 @@
 
 	&.has-background {
+		@include margin(0 auto);
 		@include padding($gutter * 2 null);
```
Your comment notes that padding will only be added to blocks without the `has-background` class, but the `_global.scss` partial has the padding set on the default class as well as the `has-background` class. Did you mean to make it global to always have padding regardless, or does it need to be removed for one of these cases?

I would think we would want the padding if a background is present so the content doesn't hit the edge of the image/color/video, but maybe I'm misinterpreting.

jomurgel replied:
Ah yes, thanks. I'll submit a few fixes and document below.

jomurgel committed:
Merge branch 'master' into hotfix/hotfix-030119-general-cleanup

jomurgel committed:
removes duplicate padding declaration

jomurgel committed:
removes additional padding on nested no-background blocks

jomurgel committed:
adds container-width margin removal to global scope

jomurgel committed:
builds

jomurgel committed:
removes container width from global scope

jomurgel committed:
adds no-background.container-width to nesting fixes

jomurgel committed:
adds posts partial for related/recent posts blocks and removes padding

jomurgel committed:
adjusts margin on container with elements

jomurgel committed:
fixes padding issues with carouel

jomurgel commented:
I've made a couple of updates for this PR based on the comment above.

First, you were right, there was some duplicate padding with the global content blocks partial. I instead  made a couple of changes.
- Replaced the duplicate padding with a margin 0 auto to remove margins if there is a background to nest properly.
- Added a margin to container-width elements so that the nesting of blocks isn't weird with white on both sides.
- Removed the padding from no-background elements if nested together: https://github.com/WebDevStudios/wd_s/blob/hotfix/hotfix-030119-general-cleanup/assets/sass/modules/content-blocks/_global.scss#L30

Tested all permutations with background, no background, full/container widths and I think this resolves any "double padding" or "no space" issues that Iw as trying to fix initially.

This also included adding the padding from the bottom of the recent posts and related posts blocks if there is no background. which "normalizes" the space between blocks instead of having an extra-large space at the bottom with the card margins.

Here's a screenshot of all of those elements/blocks nested together with the variations.

![screencapture-192-168-1-237-3000-acf-blocks-2019-03-01-09_54_12](https://user-images.githubusercontent.com/5230729/53653280-38b06500-3c08-11e9-8847-e51744136a2a.png)

coreymcollins commented on assets/sass/modules/content-blocks/_global.scss:
```diff
@@ -25,6 +26,27 @@
 	} // &.has-background
 
+	// Remove the extra padding on nested no-background blocks.
+	&.no-background,
```
Can we combine these two declarations + the one on line 41? It looks like they're all targeting the same children, so we could possibly save a few lines there if it doesn't cause any unexpected issues.

jomurgel replied:
_backs away_

jomurgel committed:
consolodates no-background padding styles

jomurgel commented:
🤦‍♀️ Consolidated the two code blocks since they use the same children. Confirmed working for me locally now.
